packages/amplify-category-function/src/events/prePushHandler.ts

```diff
@@ -11,8 +12,21 @@ import { ensureEnvironmentVariableValues } from '../provider-utils/awscloudforma
  * prePush Handler event for function category
  */
 export const prePushHandler = async (context: $TSContext): Promise<void> => {
-  await ensureEnvironmentVariableValues(context);
-  await ensureFunctionSecrets(context);
+  // if appId and envName can be resolved, proceed with checking env vars and secrets
+  const envName = stateManager.getCurrentEnvName() || context?.exeInfo?.inputParams?.amplify?.envName;
+  // get appId from amplify-meta or fallback to input params
+  const appId: string | undefined =
+    (stateManager.getMeta(undefined, { throwIfNotExist: false }) || {})?.providers?.awscloudformation?.AmplifyAppId ||
+    context?.exeInfo?.inputParams?.amplify?.appId;
+  if (envName && appId) {
+    await ensureEnvironmentVariableValues(context, appId);
+    await ensureFunctionSecrets(context);
+  } else {
+    printer.warn(
+      'Could not resolve either appId, environment name or both. Skipping environment check for function secrets and environment variables',
```

Comment:
Does this warning need to be actionable for users?

Reply:
I could add a line about "make sure your project is initialized". But it might just lead to confusion because this would only be a problem if the environment is missing some parameters

Reply:
Maybe this should be a debug line instead?